ITensorGaussianMPS/src/gmps.jl

```diff
@@ -181,82 +171,61 @@ function quadratic_hamiltonian(os::OpSum)
     sites[n] = ntuple(n -> ITensors.site(term[n]), Val(2))
     nsites = max(nsites, maximum(sites[n]))
   end
-  quadrants_present = unique(quads)
-  # currently compatible formats:
-  # [(2,2)] -> hopping only
-  # [(1,2),(2,2)] -> hopping and pairing
-  @show quadrants_present
   ElT = all(isreal(coefs)) ? Float64 : ComplexF64
-  # account for both pairing+hopping, and pairing only
-  if (Set([(2, 2), (1, 2)]) == Set(quadrants_present)) ||
-    (Set([(1, 2)]) == Set(quadrants_present))
-    println("quadratic hamiltonian with pairing")
-    nsites *= 2
-    h = zeros(ElT, nsites, nsites)
-    for n in 1:nterms
-      quad = quads[n]
-      offsets = div(nsites, 2) .* (quad .- 1)
-      h[(sites[n] .+ offsets)...] += coefs[n]
-    end
-    N = div(nsites, 2)
-    # fill in the other quadrants
-    h[1:N, 1:N] .= -conj.(h[(N + 1):end, (N + 1):end])
-    h[(N + 1):end, 1:N] .= -conj.(h[1:N, (N + 1):end])
-    h = ConservesNfParity(h)  ###storage in blocked format here
-  elseif [(2, 2)] == quadrants_present
-    println("quadratic hamiltonian without pairing")
-    h = zeros(ElT, nsites, nsites)
-    for n in 1:nterms
-      h[sites[n]...] += coefs[n]
-    end
-    h = ConservesNf(h)
-  else
-    error(
-      "Either pass all terms (for non-number conserving) or only (Cdag,C) ones (number-conserving)",
-    )
+  h = zeros(ElT, 2*nsites, 2*nsites)
+  other_quad = i -> i==2 ? 1 : 2
+  for n in 1:nterms
+    quad = quads[n]
+    offsets = nsites .* (quad .- 1)
+    h[(sites[n] .+ offsets)...] += coefs[n]
+    other_offsets = nsites .* (other_quad.(quad) .- 1)
+    h[(sites[n] .+ other_offsets)...] += -conj(coefs[n])
   end
-  return h
+  return Hermitian(interleave(h))
```

Comment:
Should we check that it is approximately Hermitian? We could use `ishermitian` and `IsApprox.jl` (https://github.com/jlapeyre/IsApprox.jl).